backend/testfiles/execution/stdlib/json.dark

```diff
@@ -403,6 +403,113 @@ module UInt16 =
     (Builtin.Json.parse<UInt16> "\"42\n\"") |> err = Result.Error "Not JSON"
 
 
+module Int128 =
+  module Basic =
+    Builtin.Json.serialize<Int128> 0Q = "0"
+    Builtin.Json.serialize<Int128> 12345Q = "12345"
+    Builtin.Json.serialize<Int128> -12345Q = "-12345"
+    Builtin.Json.parse<Int128> "0" = Result.Ok 0Q
+    Builtin.Json.parse<Int128> "0.0" = Result.Ok 0Q
+    Builtin.Json.parse<Int128> "-0.0" = Result.Ok 0Q
+    Builtin.Json.parse<Int128> "-1.0" = Result.Ok(-1Q)
+    Builtin.Json.parse<Int128> "1.2E2" = Result.Ok 120Q
+    Builtin.Json.parse<Int128> "1.2E+2" = Result.Ok 120Q
+    Builtin.Json.parse<Int128> "1200E-1" = Result.Ok 120Q
+
+  module Int128Limits =
+    Builtin.Json.serialize<Int128> -170141183460469231731687303715884105728Q = "-170141183460469231731687303715884105728"
+    Builtin.Json.serialize<Int128> 170141183460469231731687303715884105727Q = "170141183460469231731687303715884105727"
+
+    Builtin.Json.parse<Int128> "-170141183460469231731687303715884105728" = Result.Ok
+      -170141183460469231731687303715884105728Q
+
+    Builtin.Json.parse<Int128> "170141183460469231731687303715884105727" = Result.Ok
+      170141183460469231731687303715884105727Q
+
+  module Errors =
+    (Builtin.Json.parse<Int128> " ") |> err = Result.Error "Not JSON"
+    (Builtin.Json.parse<Int128> "4a") |> err = Result.Error "Not JSON"
+    (Builtin.Json.parse<Int128> "- 42") |> err = Result.Error "Not JSON"
+
+    (Builtin.Json.parse<Int128> "null") |> err = Result.Error
+      "Can't parse JSON `null` as type `Int128` at path: `root`"
+
+    (Builtin.Json.parse<Int128> "true") |> err = Result.Error
+      "Can't parse JSON `true` as type `Int128` at path: `root`"
+
+    (Builtin.Json.parse<Int128> "false") |> err = Result.Error
+      "Can't parse JSON `false` as type `Int128` at path: `root`"
+
+    (Builtin.Json.parse<Int128> "\"42\"") |> err = Result.Error
+      "Can't parse JSON `\"42\"` as type `Int128` at path: `root`"
+
+    (Builtin.Json.parse<Int128> "[42]") |> err = Result.Error
+      "Can't parse JSON `[42]` as type `Int128` at path: `root`"
+
+    (Builtin.Json.parse<Int128> "{ \"key\": 42 }") |> err = Result.Error
+      "Can't parse JSON `{ \"key\": 42 }` as type `Int128` at path: `root`"
+
+    (Builtin.Json.parse<Int128> "42.5") |> err = Result.Error
+      "Can't parse JSON `42.5` as type `Int128` at path: `root`"
+
+    (Builtin.Json.parse<Int128> "\"42\n\"") |> err = Result.Error "Not JSON"
+
+    // (-1.7014118346046923E+38) this result is out of range for Int128 and is wrapped around to the min value
+    Builtin.Json.parse<Int128> "-170141183460469231731687303715884105800" = Result.Ok
+      -170141183460469231731687303715884105728Q
```

Comment:
Not sure why it wraps around for negative out-of-range values instead of returning an out-of-range error, as it does for positive out-of-range values

Reply:
I recall adding a check for this during parsing for some integers, though I could be wrong?

Reply:
I think what you did for int64 is different if you are referring to the following comment

> // dotnet will wrap 9223372036854775808 to be -9223372036854775808 instead, we
> // don't want that and will error instead

This one only happens with negative out-of-range values and it always wraps them around to the Int128 Min value. (I could be misunderstanding something though)